app-crypt/certbot/certbot-3.2.0-r103.ebuild

```
@@ -147,161 +147,92 @@ RDEPEND="
 	done
-
-	# Restore S
-	S="${S_BACKUP}"
-
-	use doc && {
-		# Replace HTML_DOCS with one single entry to avoid merging
-		HTML_DOCS=( "${temp_docs}" )
-	}
 }
 
-python_compile_all() {
-	# There is no documentation in certbot-apache or certbot-nginx.
-	if [[ "${dir}" = "certbot-apache" ]] || [[ "${dir}" = "certbot-nginx" ]]; then
-		return
-	fi
-
-	# Used to build documentation
-	use doc && {
-		sphinx_compile_all
-
-		# Subdirectory "_build/html" from build_sphinx in eclass/python-utils-r1.eclass
-		mv "${_DISTUTILS_SPHINX_SUBDIR}/_build/html" "${temp_docs}/${dir}" || die
-	}
-}
+# # Used to build documentation
+# python_compile_all() {
+# 	use doc || return
+
+# 	local dir
+# 	for dir in "${CERTBOT_DIRS[@]}"; do
+# 		# There is no documentation in certbot-apache or certbot-nginx.
+# 		if [[ "${dir}" = "certbot-apache" ]] || [[ "${dir}" = "certbot-nginx" ]]; then
+# 			continue
+# 		fi
+
+# 		pushd "${dir}" > /dev/null || die
+
+# 		# Reset HTML_DOCS between sphinx calls as we discard its content.
+# 		local HTML_DOCS=()
+
+# 		sphinx_compile_all
+
+# 		# Note: directory "${dir}" does not exist, so it will fail if
+# 		# HTML_DOCS has multiple entries.
+# 		# Note: can’t use following instruction, because of entries
+# 		# ending with `/.`, see build_sphinx in python-utils-r1 eclass.
+# 		# ```
+# 		# mv "${HTML_DOCS[@]}" "${CERTBOT_DOCS}/${dir}" || die
+# 		# ```
+# 		# mv: cannot move 'docs/_build/html/.' to '[…]/temp/docs/acme': Device or resource busy
+# 		# Works without ending dot. Workaround applied.
+# 		local doc_dir
+# 		for doc_dir in "${HTML_DOCS[@]}"; do
+# 			# Let’s ignore  suffixe `/.` in `docs/_build/html/.`.
+# 			mv "${doc_dir%/.}" "${CERTBOT_DOCS}/${dir}" || die
+# 		done
+
+# 		popd > /dev/null || die
+# 	done
+
+# 	# And finally give the result.
+# 	HTML_DOCS=( "${CERTBOT_DOCS}" )
+# }
 
 python_test() {
 	local -x PYTEST_DISABLE_PLUGIN_AUTOLOAD=1
-	epytest
-}
-
-src_test() {
-	local S_BACKUP="${S}"
-
-	local certbot_dirs=()
-	local base module dir
-	for base in "${CERTBOT_BASE[@]}"; do
-		certbot_dirs+=("${base}")
-	done
-	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
-		use "certbot-${module}" \
-			&& certbot_dirs+=("certbot-${module}")
-	done
-
-	for dir in "${certbot_dirs[@]}"; do
-		S="${WORKDIR}/${P}/${dir}"
-		pushd "${S}" > /dev/null || die
-		distutils-r1_src_test
-		popd > /dev/null || die
-	done
-
-	# Restore S
-	S="${S_BACKUP}"
-}
-
-src_install() {
-	local S_BACKUP="${S}"
-
-	local certbot_dirs=()
-	local base module dir
-	for base in "${CERTBOT_BASE[@]}"; do
-		certbot_dirs+=("${base}")
-	done
-	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
-		use "certbot-${module}" \
-			&& certbot_dirs+=("certbot-${module}")
-	done
-
-	for dir in "${certbot_dirs[@]}"; do
-		S="${WORKDIR}/${P}/${dir}"
-		pushd "${S}" > /dev/null || die
-		distutils-r1_src_install
-		popd > /dev/null || die
-	done
-
-	# Restore S
-	S="${S_BACKUP}"
+	# Argument required to change pytest rootdir.
+	epytest "${BUILD_DIR}"
```

```suggestion
	epytest "${BUILD_DIR}/install"
```

Now that we don't clean setuptools `build` dirs.

---

Thank you @mgorny, this solution fix the tests for release 4.0.0.

I will create a dedicated PR to push quickly this fix, as it was expected in PR #41504.